internal/signer/darwin/keychain/keychain.go

```diff
@@ -388,6 +406,22 @@ func identityToSecKeyRef(ident C.SecIdentityRef) (C.SecKeyRef, error) {
 	return ref, nil
 }
 
+func identityToPublicSecKeyRef(ident C.SecIdentityRef) (C.SecKeyRef, error) {
+	var key C.SecKeyRef
+	var certRef C.SecCertificateRef
+	if errno := C.SecIdentityCopyCertificate(ident, &certRef); errno != 0 {
+		return 0, keychainError(errno)
+	}
+	defer C.CFRelease(C.CFTypeRef(certRef))
+
+	key = C.SecCertificateCopyKey(certRef)
+
+	if key == 0 {
+		return 0, fmt.Errorf("Public key was NULL")
```

Comment:
I try to write my error messages in the following format:

"What went wrong. Why it went wrong. What could be the reason"

This makes future issues easier to debug.

Reply:
Ahh, got it I'll go ahead and update my error messages